### ISSUE TITLE
$this autocompletion in macros

### ISSUE THREAD

**Issue by olivernybroe** [laravel-idea/plugin#345]
When defining a macro like the following
```php
Route::macro('coolAction', function (): bool {
	return $this->getAction('cool');
});
```

The autocompletion for `$this` does not work inside of the macro 😄 

Fortunately all Laravel macros are using the `Macroable` trait which makes it easier to make a general solution and the class `$this` will a type of will always be the class which the `macro` static method is called on 👍 

One other is the Carbon library though, they have build their own macroable logic, so to also supports macros in their class you might need a special case for them.

**Comment by adelf** [comment]
It sounds close to what I did for PEST plugin)
I think I can do it. Thanks.

**Comment by olivernybroe** [comment]
Exactly 🙂 only that it will be simpler as you don't have to find out what class it will be from other files.

I exactly had to do something similar in another place in pest where I reused a lot of the same code.

**Comment by adelf** [comment]
Implemented in the 4.4 version.
Full list of changes - https://laravel-idea.com/blog/2021/4-4-release
